site/content/docs/03-compile-time.md

```diff
@@ -0,0 +1,256 @@
+---
+title: Compile time
+---
+
+Typically, you won't interact with the Svelte compiler directly, but will instead integrate it into your build system using a bundler plugin:
+
+* [rollup-plugin-svelte](https://github.com/rollup/rollup-plugin-svelte) for users of [Rollup](https://rollupjs.org)
+* [svelte-loader](https://github.com/sveltejs/svelte-loader) for users of [webpack](https://webpack.js.org)
+* [parcel-plugin-svelte](https://github.com/DeMoorJasper/parcel-plugin-svelte) for users of [Parcel](https://parceljs.org/)
+
+Nonetheless, it's useful to understand how to use the compiler, since bundler plugins generally expose compiler options to you.
+
+
+
+### svelte.compile
+
+```js
+result: {
+	js,
+	css,
+	ast,
+	warnings,
+	vars,
+	stats
+} = svelte.compile(source: string, options?: {...})
+```
+
+| option | type | default
+| --- | --- | --- |
+| `filename` | string | `null`
+| `name` | string | `null`
+| `format` | `"esm"` or `"cjs"` | `"esm"`
+| `generate` | `"dom"` or `"ssr"` | `"dom"`
+| `dev` | boolean | `false`
+| `immutable` | boolean | `false`
+| `hydratable` | boolean | `false`
+| `legacy` | boolean | `false`
+| `customElement` | boolean | `false`
+| `css` | boolean | `true`
+| `preserveComments` | boolean | `false`
+| `preserveWhitespace` | boolean | `false`
+| `outputFilename` | string | `null`
+| `cssOutputFilename` | string | `null`
+| `sveltePath` | string | `"node_modules/svelte"`
```

Comment:
Saying the default is `"node_modules/svelte"` isn't really accurate. That's probably where the imports resolve to, but the path in the compiled code is just going to default to `"svelte"`.